src/Cardano/Wallet/CoinSelection.hs

```diff
@@ -0,0 +1,154 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE RankNTypes #-}
+
+{-# OPTIONS_GHC -fno-warn-redundant-constraints #-}
+
+-- |
+-- Copyright: © 2018-2019 IOHK
+-- License: MIT
+--
+-- Provides the API of Coin Selection algorithm and Fee related functionality.
+-- For more information refer to:
+-- https://iohk.io/blog/self-organisation-in-coin-selection/
+
+
+module Cardano.Wallet.CoinSelection where
+
+import Prelude
+
+import Cardano.Wallet.Primitive.Types
+    ( Coin, TxIn, TxOut, UTxO )
+import Control.DeepSeq
+    ( NFData )
+import Control.Monad.Random
+    ( MonadRandom )
+import Data.ByteString
+    ( ByteString )
+import Data.Fixed
+    ( Nano )
+import Data.List.NonEmpty
+    ( NonEmpty (..) )
+import Data.Text
+    ( Text )
+import Data.Word
+    ( Word64 )
+import GHC.Generics
+    ( Generic )
+
+data ExpenseRegulation =
+      SenderPaysFee
+      -- ^ The sender pays for the fee. This is the typical case.
+    | ReceiverPaysFee
```

Comment:
This branch (and whole `ExpenseRegulation` type) can go, we only consider `SenderPaysFee`.

Reply:
removed `ExpenseRegulation`